File: api/verify.py
from http.server import BaseHTTPRequestHandler
import json
import os

class handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
    
    def do_POST(self):
        try:
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length)
            data = json.loads(post_data)
            
            key = data.get('key', '')
            hwid = data.get('hwid', '')
            
            # Load keys
            with open('keys.json', 'r') as f:
                keys_data = json.load(f)
            
            # Validate
            if not key or not hwid:
                self.send_error(400, 'Missing key or HWID')
                return
                
            if key not in keys_data['keys']:
                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'valid': False, 'message': 'Invalid key'}).encode())
                return
                
            if key in keys_data['disabled_keys']:
                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'valid': False, 'message': 'Key disabled'}).encode())
                return
            
            key_data = keys_data['keys'][key]
            
            if key_data['hwid'] != hwid:
                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'valid': False, 'message': 'HWID mismatch'}).encode())
                return
                
            # Success
            self.send_response(200)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({
                'valid': True,
                'message': 'Key verified successfully',
                'user_name': key_data['user_name'],
                'generated_at': key_data['generated_at']
            }).encode())
            
        except Exception as e:
            self.send_error(500, f'Server error: {str(e)}')

File: api/verify.js
// Load keys from the same file your bot uses
const keys = require('../keys.json');

export default async function handler(req, res) {
  // Enable CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  // Handle preflight requests
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  // Only allow POST requests
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const { key, hwid } = req.body;

    // Validate input
    if (!key || !hwid) {
      return res.status(400).json({ 
        valid: false, 
        message: 'Missing key or HWID' 
      });
    }

    // Check if key exists
    if (!keys.keys[key]) {
      return res.json({ 
        valid: false, 
        message: 'Invalid key' 
      });
    }

    // Check if key is disabled
    if (keys.disabled_keys.includes(key)) {
      return res.json({ 
        valid: false, 
        message: 'Key disabled by admin' 
      });
    }

    const keyData = keys.keys[key];

    // Check HWID match (exact match required)
    if (keyData.hwid !== hwid) {
      return res.json({ 
        valid: false, 
        message: 'HWID mismatch' 
      });
    }

    // Check if key is enabled
    if (keyData.enabled === false) {
      return res.json({ 
        valid: false, 
        message: 'Key disabled' 
      });
    }

    // Success!
    return res.json({
      valid: true,
      message: 'Key verified successfully',
      user_name: keyData.user_name,
      generated_at: keyData.generated_at,
      user_id: keyData.user_id
    });

  } catch (error) {
    console.error('Verification error:', error);
    return res.status(500).json({ 
      valid: false, 
      message: 'Internal server error' 
    });
  }
}
